backend/routes/user.js

```diff
@@ -0,0 +1,124 @@
+const express = require('express');
+const router = express.Router();
+const gravatar = require('gravatar');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const passport = require('passport');
+const validateRegisterInput = require('../validation/register');
+const validateLoginInput = require('../validation/login');
+
+const User = require('../models/User');
+
+router.post('/register', function(req, res) {
+
+  const {
+    errors,
+    isValid
+  } = validateRegisterInput(req.body);
+
+  if (!isValid) {
+    console.log("error in 20 lines of code user.js")
+    return res.status(400).json(errors);
+  }
+  User.findOne({
+    email: req.body.email
+  }).then(user => {
+    if (user) {
+      return res.status(400).json({
+        email: 'Email already exists'
+      });
+    } else {
+      const avatar = gravatar.url(req.body.email, {
+        s: '200',
+        r: 'pg',
+        d: 'mm'
+      });
+      const newUser = new User({
+        name: req.body.name,
+        email: req.body.email,
+        password: req.body.password,
+        avatar
+      });
+
+      bcrypt.genSalt(10, (err, salt) => {
+        if (err) console.error('There was an error', err);
+        else {
+          bcrypt.hash(newUser.password, salt, (err, hash) => {
+            if (err) console.error('There was an error', err);
+            else {
+              newUser.password = hash;
+              newUser
+                .save()
+                .then(user => {
+                  res.json(user)
+                });
+            }
+          });
+        }
+      });
+    }
+  });
+});
+
+router.post('/login', (req, res) => {
```

Comment:
too long callback function

For example, create some controls folder, create login controller and move there callback in separate function.

split function. For example,
call to backend (if email exist)
register bussines logic.

Reply:
done